crates/worker/src/partition/snapshots/repository.rs

```diff
@@ -405,6 +405,10 @@ impl SnapshotRepository {
             return Ok(None); // perhaps this needs to be a configuration error
         }
 
+        if !self.staging_dir.exists() {
+            std::fs::create_dir_all(&self.staging_dir)?;
+        }
+
```

Comment:
Would this make more sense if moved to `create_if_configured` ?

Reply:
I deliberately put it closer to the point where we need to use it; a couple of reasons:

- we won't create it if a snapshots destination is configured, but the node only ever produces snapshots, and never downloads any
- we will (re-)create it just-in-time, in case someone or something helpfully deletes the empty staging directory created on startup